src/nfa.rs

```diff
@@ -55,39 +55,39 @@ pub enum CharacterClass {
 }
 
 impl CharacterClass {
-    pub fn any() -> CharacterClass {
-        Ascii(u64::MAX, u64::MAX, true)
+    pub fn any() -> Self {
+        Ascii(u64::max_value(), u64::max_value(), true)
     }
 
-    pub fn valid(string: &str) -> CharacterClass {
-        ValidChars(CharacterClass::str_to_set(string))
+    pub fn valid(string: &str) -> Self {
+        ValidChars(Self::str_to_set(string))
     }
 
-    pub fn invalid(string: &str) -> CharacterClass {
-        InvalidChars(CharacterClass::str_to_set(string))
+    pub fn invalid(string: &str) -> Self {
+        InvalidChars(Self::str_to_set(string))
     }
 
-    pub fn valid_char(char: char) -> CharacterClass {
+    pub fn valid_char(char: char) -> Self {
         let val = char as u32 - 1;
 
         if val > 127 {
-            ValidChars(CharacterClass::char_to_set(char))
+            ValidChars(Self::char_to_set(char))
         } else if val > 63 {
             Ascii(1 << (val - 64), 0, false)
         } else {
             Ascii(0, 1 << val, false)
         }
     }
 
-    pub fn invalid_char(char: char) -> CharacterClass {
+    pub fn invalid_char(char: char) -> Self {
         let val = char as u32 - 1;
 
         if val > 127 {
-            InvalidChars(CharacterClass::char_to_set(char))
+            InvalidChars(Self::char_to_set(char))
         } else if val > 63 {
-            Ascii(u64::MAX ^ (1 << (val - 64)), u64::MAX, true)
+            Ascii(u64::max_value() ^ (1 << (val - 64)), u64::max_value(), true)
```

Comment:
Just wondering - so I learn the difference here - is there a reason for switching from using the constant to the method?

Reply:
It avoids having to import the `std::u64` module, and there is [a plan to deprecate that module and replace it with associated constants](https://github.com/rust-lang/rfcs/pull/2700).